__tests__/app.test.js

```diff
@@ -98,3 +98,215 @@ describe("GET /api/articles", () => {
+      inc_votes: 5,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(patchVote)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: 105,
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+  test("update article with object inc_votes: newVote, returns updated article with negative number value", () => {
+    const patchVote = {
+      inc_votes: -150,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(patchVote)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: -50,
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+  test("Responds with the correct error code when username does not exist", () => {
+    const patchVote = {
+      inc_votessss: 5,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(patchVote)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid column value");
+      });
+  });
+});
+
+describe("DELETE /api/comments/:comment_id", () => {
+  test("responds with status 204 and no content", () => {
+    return request(app)
+    .delete("/api/comments/3")
+    .expect(204)
+  })
+  test("responds with status 404 when the comment to delete doesn't exist", () => {
+    return request(app)
+    .delete("/api/comments/39999")
+    .expect(404)
+    .then(({body}) => {
+      expect(body.msg).toBe("Comment not found")
+    })
+  })
+})
+
+describe("GET /api/users", () => {
+  test("responds with 200 and an array of object with the correct properties", () => {
+    return request(app)
+    .get("/api/users")
+    .expect(200)
+    .then(({body}) => {
+      expect(body).toHaveLength(4);
+      body.forEach((user) => {
+      expect(user).toMatchObject({
+        username: expect.any(String),
+        name: expect.any(String),
+        avatar_url: expect.any(String)
+      });
+    })
+  })
+})
+})
+
+describe("GET /api/articles", () => {
+  test("GET 200 responds with articles filtered by topic", () => {
+    return request(app)
+    .get("/api/articles?topic=mitch")
+    .expect(200)
+    .then(({body}) => {
+      const { articles } = body
+      expect(articles).toHaveLength(12);
+      articles.forEach((article) => {
+      expect(article).toMatchObject({
+        title: expect.any(String),
+        topic: "mitch",
+        author: expect.any(String),
+        created_at: expect.any(String),
+        article_img_url: expect.any(String)
```

Comment:
We get more properties back than this I believe? comment_count, votes and article_id?